framework/src/components/system/system.js

```diff
@@ -114,37 +114,35 @@ class System {
 	/**
 	 * Updates private broadhash and height values.
 	 *
-	 * @param {function} cb - Callback function
-	 * @returns {setImmediateCallback} cb, err
+	 * @returns {Promise.<boolean, Error>}
```

Comment:
it should be `Promise<boolean | Error>`, but I think error should be handled in each places (it seems like anyway it's doing so)
So prefer to have Promise<void> here

Reply:
I've followed the same pattern suggested by @lsilvs . It is been used in storage entities.